src/neo-vm/Utility.cs

```diff
@@ -25,6 +25,26 @@ static Utility()
             StrictUTF8.EncoderFallback = EncoderFallback.ExceptionFallback;
         }
 
+        public static BigInteger ModInverse(this BigInteger n, BigInteger modulus)
+        {
+            if (n <= 0) throw new ArgumentOutOfRangeException(nameof(n));
+            if (modulus < 2) throw new ArgumentOutOfRangeException(nameof(modulus));
+            BigInteger m = modulus, v = 0, d = 1;
+            while (n > 0)
+            {
+                BigInteger t = m / n, x = n;
+                n = m % x;
+                m = x;
+                x = d;
+                d = checked(v - t * x);
```

Comment:
```suggestion
                d = v - t * x;
```

By the way, I don't think we need this `checked` here, as we are dealing with BigInteger type, not unsafe ones,

Reply:
@shargon I also suggest you to refactor the code, renaming `x` to `aux`.. I think it helps a little bit.

Reply:
In fact, this is just crazy `x = d;  d = checked(v - t * x);`... please just use: `d = v - t * d;`